src/toil/job.py

```diff
@@ -157,11 +159,65 @@ def _parseResource(name, value):
                             % (name, type(value)))
 
 
-class Job(ResourceRequirementMixin):
+class JobLikeObject(ResourceRequirementMixin):
+    def __init__(self, name, job=None, **requirements):
+        super(JobLikeObject, self).__init__(**requirements)
+        self.name = name
+        self.job = job or self.__class__.__name__
+
+    def __str__(self):
+        printedName = '\''+self.job + '\''
+        if self.name:
+            printedName += ' '+self.name
+        return str(printedName)
+
+class JobNode(JobLikeObject):
+    """
+    This object bridges the job graph, job, and batchsystem classes
+    """
+    def __init__(self, memory, cores, disk, preemptable, job, name, jobStoreID,
+                 command, predecessorNumber=1):
+        super(JobNode, self).__init__(memory=memory, cores=cores, disk=disk,
+                                      preemptable=preemptable, name=name, job=job)
+        self.jobStoreID = jobStoreID
+        self.predecessorNumber = predecessorNumber
+        self.command = command
+
+    def __hash__(self):
+        return hash(self.jobStoreID)
+
+    def __ne__(self, other):
```

Comment:
You need to override `__eq__` to compare based on ID. This overrides `__ne__` with the default implementation.